EISW-88506 Extend G-API classification benchmark to work with different inference backends (#3853)
openvinotoolkit/open_model_zoo: demos/classification_benchmark_demo/cpp_gapi/README.md

    -no_show                  Optional. Disable showing of processed images.
    -time "<integer>"         Optional. Time in seconds to execute program. Default is -1 (infinite time).
    -u                        Optional. List of monitors to show initially.

>     -backend <string>         Optional. Specify an inference backend. The list of available backends depend on openCV version. Default value is IE.

Author: Sorry, I don't understand your point.
There is an actual output from benchmark:

```
build_ocv_2/intel64/Debug/classification_benchmark_demo_gapi -h
[ INFO ] OpenVINO
[ INFO ] 	version: 2023.0.1
[ INFO ] 	build: 2023.0.1-11048-a6351294e77-HEAD

classification_benchmark_demo [OPTION]
Options:
   ....
    -backend                  Optional. Specify an inference backend (the list of available backends: IE,ONNX,OV)
```
what kind of misalignment do you mention about?

>     -backend <string>         Optional. Specify an inference backend. The list of available backends depend on openCV version. Default value is IE.

Reviewer: You have `Default value is IE.` in README, but not in help print. You should also stay consistent about using `()` for `the list of available backends`.

Another point which didn't exist at the moment of writing the initial comment is `See README.md for details.`

>     -backend <string>         Optional. Specify an inference backend. The list of available backends depend on openCV version. Default value is IE.

Author: ```
-backend                  Optional. Specify an inference backend. The list of available backends: IE,ONNX,OV. Default value is IE. See README.md for details
```
Do you think this is ok?

>     -backend <string>         Optional. Specify an inference backend. The list of available backends depend on openCV version. Default value is IE.

Reviewer: Yes, as long as it is aligned between the implementation and README

>     -backend <string>         Optional. Specify an inference backend. The list of available backends depend on openCV version. Default value is IE.

Reviewer: ```suggestion
    -backend <string>         Optional. Specify an inference backend. The list of available backends depends on openCV version. Default value is IE. See README.md for details
```

                      -u CDM
```

> To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

Reviewer: `onnxruntime_providers_openvino.dll` is already mentioned

`DirectML.dll` doesn't exist for me. Are you sure?

> To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

Reviewer 2: `DirectML.dll` should be in binary folder if you built `DML` execution provider.

> To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

Reviewer 2: Well, in fact `DirectML.dll` is also part of `C:\system32` but if it's custom it also should be placed near the binary

> To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

Author: I suppose whether anyone has proficient knowledge how to build the custom `DirectML.dll` instead of using the default one then this experimentator is well informed how to deploy it. In my perspective, this comprehensive instructions are redundant

> To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

Reviewer: The instructions are mostly for me in first place. Usually I forget such specifics in two weeks.

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=DEBUG` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.

Reviewer 2: Was it fixed?

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=DEBUG` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.

Reviewer 2: Was it fixed

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=DEBUG` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.

Reviewer 3: > IE backend was introduced in opencv 4.5.2 version.

Actually in 4.2.0... https://github.com/opencv/opencv/wiki/ChangeLog#version420

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=DEBUG` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.

Reviewer 3: And ONNX in 4.5.1: https://github.com/opencv/opencv/wiki/ChangeLog#version451

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=DEBUG` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.

Author: Will be added

                      -u CDM
```

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8.

Reviewer: ```suggestion
To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.0, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found by running `set OPENCV_LOG_LEVEL=DEBUG` and ruining the demo with `--backend ONNX/DML/`. Search for `Available DirectML adapters:`. For ` -backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.
```

                      -u CDM
```

To let the demo find onnx libs copy `onnxruntime_providers_openvino.dll` from `lib` and `onnxruntime.dll`, `onnxruntime_providers_shared.dll` from `bin` dirs  of onnxruntume install to the folder with the demo executable.

> The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.5.2, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=INFO` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OVEP/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.

Reviewer: ```suggestion
The inference backend is specified by -backend flag. Depends on openCV version there might be different backends available: `IE` starting from openCV 4.5.2, `ONNX` from 4.7.1 and `OV` from 4.8. The possible device value for `"ONNX/DML/<device>` can be found using `set OPENCV_LOG_LEVEL=INFO` and running the demo with `-backend ONNX/DML/`. Search for `Available DirectML adapters:`. For `-backend ONNX/OV/<device>` the device is the same as the one selected for `--use_openvino` while compiling `onnxruntime`.
```
